modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -79,6 +79,8 @@ def __init__(
         self.num_triggers = 0
         self.current_training_id: Optional[int] = None
         self.triggers: list[int] = []
+        # this is to store the first and last timestamp of remaining data in _handle_triggers_within_batch
+        self.remaining_data_range: Optional[tuple[int, int]] = None
```

Comment:
So if I understand correctly, this stores the range of the leftover data from the last trigger, no?

Reply:
I am wondering if we can make this clear in the name somehow. Otherwise, could you expand the comment to say instead of just "remaining data" to be a bit more precise and say "This stores the first and last timestamp in case we have remaining data after handling all triggers in `_handle_triggers_within_batch `"

Reply:
yes you understand it correctly. I don't have a better name but will change the comment!